src/components/modal/__internal__/modal-manager.test.tsx

```diff
@@ -0,0 +1,325 @@
+  mockModal2 = document.createElement("div");
+
+  ModalManager.clearList();
+
+  ModalManager.addModal(mockModal1);
+  ModalManager.addModal(mockModal2);
+  ModalManager.removeModal(mockModal2);
+
+  expect(cb1).not.toHaveBeenCalledWith(true);
+
+  cb1?.mockClear();
+  cb2?.mockClear();
+});
+
+test("does not trigger refocus if the triggerRefocusOnClose flag passed to removeModal is set to false when the removeModal method has been called and the param matches a Modal in the list", () => {
+  cb1 = jest.fn();
+  cb2 = jest.fn();
+
+  mockModal1 = document.createElement("div");
+  mockModal2 = document.createElement("div");
+
+  ModalManager.clearList();
+
+  ModalManager.addModal(mockModal1, cb1);
+  ModalManager.addModal(mockModal2, cb2);
+  ModalManager.removeModal(mockModal2, false);
+
+  expect(cb1).not.toHaveBeenCalledWith(true);
+
+  cb1?.mockClear();
+  cb2?.mockClear();
+});
+
+test("triggers refocus if a callback is provided and the triggerRefocusOnClose flag is not passed to removeModal when the removeModal method has been called and the param matches a Modal in the list", () => {
+  cb1 = jest.fn();
+  cb2 = jest.fn();
+
+  mockModal1 = document.createElement("div");
+  mockModal2 = document.createElement("div");
+
+  ModalManager.clearList();
+
+  ModalManager.addModal(mockModal1, cb1);
+  ModalManager.addModal(mockModal2, cb2);
+
+  expect(cb1).not.toHaveBeenCalledWith(true);
+
+  ModalManager.removeModal(mockModal2);
+
+  expect(cb1).toHaveBeenCalledWith(true);
+
+  cb1?.mockClear();
+  cb2?.mockClear();
+});
+
+test("should no longer be in the global variable window.__CARBON_INTERNALS_MODAL_LIST when the removeModal method has been called and the param matches a Modal in the list", () => {
+  cb1 = jest.fn();
+  cb2 = jest.fn();
+
+  mockModal1 = document.createElement("div");
+  mockModal2 = document.createElement("div");
+
+  ModalManager.clearList();
+
+  ModalManager.addModal(mockModal1, cb1);
+  ModalManager.addModal(mockModal2, cb2);
+
+  expect(window.__CARBON_INTERNALS_MODAL_LIST?.length).toEqual(2);
+  expect(window.__CARBON_INTERNALS_MODAL_LIST?.[0]).toEqual({
+    modal: mockModal1,
+    setTriggerRefocusFlag: cb1,
+  });
+  expect(window.__CARBON_INTERNALS_MODAL_LIST?.[1]).toEqual({
+    modal: mockModal2,
+    setTriggerRefocusFlag: cb2,
+  });
+
+  ModalManager.removeModal(mockModal2);
+
+  expect(window.__CARBON_INTERNALS_MODAL_LIST?.length).toEqual(1);
+  expect(window.__CARBON_INTERNALS_MODAL_LIST?.[0]).toEqual({
+    modal: mockModal1,
+    setTriggerRefocusFlag: cb1,
+  });
+
+  ModalManager.removeModal(mockModal1);
+
+  expect(window.__CARBON_INTERNALS_MODAL_LIST?.length).toEqual(0);
+
+  cb1?.mockClear();
+  cb2?.mockClear();
+});
+
+// This test case is required to hit coverage. It covers the case where the modal to be removed is not in the `modalList` array.
+test("when the removeModal method has been called and the param does not match a Modal in the list, then nothing happens", () => {
```

Comment:
I've removed the comment. I had this here because I struggled for a few hours to get the code coverage for the particular lines of code this covered. I put this comment here so someone in the future wouldn't have the same issue I did 🤣